forage-android/src/main/java/com/joinforage/forage/android/ecom/services/vault/vgs/VGSResponseParser.kt

```diff
@@ -0,0 +1,74 @@
+package com.joinforage.forage.android.ecom.services.vault.vgs
+
+import com.joinforage.forage.android.core.services.forageapi.network.ForageApiError
+import com.joinforage.forage.android.core.services.forageapi.network.ForageApiResponse
+import com.joinforage.forage.android.core.services.forageapi.network.ForageError
+import com.joinforage.forage.android.core.services.forageapi.network.UnknownErrorApiResponse
+import com.joinforage.forage.android.core.services.vault.VaultResponseParser
+import com.verygoodsecurity.vgscollect.core.model.network.VGSResponse
+import org.json.JSONException
+
+class VGSResponseParser(vgsRes: VGSResponse?) : VaultResponseParser {
+    override val isNullResponse: Boolean
+    override val vaultError: ForageApiResponse.Failure?
+    override val forageError: ForageApiResponse.Failure?
+    override val successfulResponse: ForageApiResponse.Success<String>?
+
+    override val vaultErrorMsg: String = vgsRes?.body.toString()
+    override val rawResponse: String = vgsRes.toString()
+
+    init {
+        if (vgsRes == null) {
+            isNullResponse = true
+            vaultError = null
+            forageError = null
+            successfulResponse = null
+        } else {
+            isNullResponse = false
+            vaultError = parseVaultError(vgsRes)
+            forageError = parseForageError(vgsRes)
+            successfulResponse = parseSuccessfulResponse(vgsRes)
+        }
+    }
+
+    private fun parseVaultError(res: VGSResponse): ForageApiResponse.Failure? {
+        if (res is VGSResponse.SuccessResponse) return null
+
+        // given that it's not a success and it's not a forage error
+        // that only leaves a VGS error.
+        // TODO: investigate how to get identify the actual VGS error
+        //  for logging purposes
+        return if (parseForageError(res) == null) UnknownErrorApiResponse else null
```

Comment:
Great catch, [seems to be `null` now ](https://app.datadoghq.com/logs?query=status:error%20env:(prod%20OR%20sandbox)%20service:android-sdk%20%22Received%20error%20from%22%20&agg_m=count&agg_m_source=base&agg_t=count&cols=host,service&fromUser=true&messageDisplay=inline&sort=time&storage=hot&stream_sort=desc&view=spans&viz=stream&from_ts=1713732669639&to_ts=1714848398962&live=false) 😅